frappe/templates/includes/login/login.js

```diff
@@ -21,7 +45,7 @@ login.bind_events = function () {
 		args.usr = frappe.utils.xss_sanitise(($("#login_email").val() || "").trim());
 		args.pwd = $("#login_password").val();
 		if (!args.usr || !args.pwd) {
-			frappe.msgprint('{{ _("Both login and password required") }}');
+			frappe.msgprint(window.strings.missing_values);
```

Comment:
```suggestion
			frappe.msgprint('{{ _("Both login and password required") | e }}');
```


Does this work?

Reply:
https://jinja.palletsprojects.com/en/3.0.x/templates/#working-with-manual-escaping

Reply:
`json` doesn't seem like valid escape. We are escaping for HTML right?

Reply:
> `json` doesn't seem like valid escape

It can be a valid escape because the resulting value is parsed by the JavaScript parser inside the script tag. The `json` Jinja filter produces a valid JS primitive value/object/array.

On the other hand, `e` escapes values as HTML entities, which might not work everywhere.

Does `msgprint` escape (with HTML entities) the string it receives before putting it in a dialog (or uses `$.text(...)`)?

```jinja
{{ 'escape"\'<>' | json }} => "escape\"'<>"

"{{ 'escape"\'<>' | e }}" => "escape&#34;&#39;;&lt;&gt;"
```

Reply:
In this context I am guessing JSON will work too. 

It's quite annoying to add it for string though. But ig we don't have an alternate.

Reply:
It can work but there are still some issues, because the encoding strategy depends on the consumer of the string:

```
jQuery.text() -> json
jQuery.html() -> e
frappe.msgprint() -> ?
login.set_status() -> ?
```

Currently, because every string in stored in `window.strings` and JSON-encoded, it remains HTML injection issues when the translated string contains `<>`.

**It's the same situation as with `frappe._messages` on the desk/web, I kinda removed Jinja from the loop.**